### PR TITLE
NMS-16530: OpenNMS RESTful API provides incorrect URL in GUI after SSL enablement.

### PR DESCRIPTION
**Pull request by Junaid-Khan-Nant**
- https://opennms.atlassian.net/browse/NMS-16530

### CODE REVIEW

File: ui/src/containers/OpenAPI.vue [hunk 1]
```diff
@@ -80,11 +80,20 @@ const setup = async () => {
 
   if (protocol === https) {
     const openApiSpecString = JSON.stringify(openApiSpec)
-    const modifiedOpenApiSpecString = openApiSpecString.replaceAll(http, https)
+    let modifiedOpenApiSpecString;
+    if (!(openApiSpecString.includes("https"))) {
+      modifiedOpenApiSpecString= openApiSpecString.replaceAll(http, https)
+    }else{
+    modifiedOpenApiSpecString = openApiSpecString
+    }
```
Comment (cgorantla) [inline ui/src/containers/OpenAPI.vue:88]: can replace with ternary operator to simplify this.

**Review by cgorantla — COMMENTED** [review]
Good find.

**Review by smunir-onms — COMMENTED** [review]
LGTM, please take care of the formating (spotless:apply manually)

File: ui/src/containers/OpenAPI.vue [hunk 2]
```diff
@@ -80,11 +80,16 @@ const setup = async () => {
 
   if (protocol === https) {
     const openApiSpecString = JSON.stringify(openApiSpec)
-    const modifiedOpenApiSpecString = openApiSpecString.replaceAll(http, https)
+    let modifiedOpenApiSpecString;
+    modifiedOpenApiSpecString = openApiSpecString.includes("https")
```
[the remark this answers is not in the document]
Comment (Junaid-Khan-Nant) [reply ui/src/containers/OpenAPI.vue]: Done.

File: ui/src/containers/OpenAPI.vue [hunk 3]
```diff
@@ -80,11 +80,16 @@ const setup = async () => {
...
+      ? openApiSpecString
+      : openApiSpecString.replaceAll(http, https);
     modifiedOpenApiSpec = JSON.parse(modifiedOpenApiSpecString)
-
     const openApiSpecStringV1 = JSON.stringify(openApiSpecV1)
-    const modifiedOpenApiSpecStringV1 = openApiSpecStringV1.replaceAll(http, https)
+    let modifiedOpenApiSpecStringV1;
+    modifiedOpenApiSpecStringV1 = openApiSpecStringV1.includes("https")
```
[the remark this answers is not in the document]
Comment (Junaid-Khan-Nant) [reply ui/src/containers/OpenAPI.vue]: Done.

**Review by synqotik — APPROVED** [review]
LGTM

**Review by cgorantla — APPROVED** [review]
LGTM!

**Pull request merged**